packages/std/src/coin.rs

```diff
@@ -55,6 +55,31 @@ impl fmt::Display for Coin {
     }
 }
 
+/// Equivalent to Coin, but with a reference to the denom
+#[derive(Serialize, Deserialize, Debug, Clone, Default, PartialEq, Eq, JsonSchema)]
+pub struct CoinRef<'a> {
+    pub denom: &'a str,
+    pub amount: Uint128,
+}
+
+impl From<CoinRef<'_>> for Coin {
```

Comment:
I think this should better be implemented with `ToOwned`

Reply:
Unfortunately, we cannot implement `ToOwned`, because it would require the owned type (`Coin`) to implement `Borrow<CoinRef>`. To do that, we would have to implement a function that takes a `&Coin` and returns a `&CoinRef`, which is not possible.